packages/backend/src/core/DataSyncService.ts

```diff
@@ -46,10 +62,42 @@ export class DataSyncService {
       stateTransitionFacts.map((f) => f.hash)
     )
 
-    for (const { pages } of stateTransitions) {
+    let { oldHash, id } = await this.readLastUpdate()
```

Comment:
Maybe this part of the `sync` method (from this line downwards) could be extracted into it's own class? It introduces 3 new dependencies to `DataSyncService` and is also a bit hard to test. If we instead extract it it would be a lot easier to manage.

Reply:
Good idea!